File: script_linux/21/exemple_2.py
#! /usr/bin/python

def premier (n) :
	i = n - 1
	n_premier = 0
	while i > 1 :
		if ((n % i) == 0) : break;
		i = i - 1
	else : n_premier = 1
	return (n_premier)

j = 1
while (j < 500) :
	if (premier (j)) :
		print j,
	j = j + 1

File: script_linux/22/exemple_2.rb
#! /usr/local/bin/ruby

class Nouvelle

	def initialize (num)
		@numero = num
	end

	def publique_1 ()
		print "Instance ", @numero, " : methode publique 1\n"
	end

	def publique_2 (instance)
		print "Instance ", @numero, " : methode publique 2\n"
		print "    invocation de protegee()\n"
		instance . protegee ()
	end

	def publique_3 ()
		print "Instance ", @numero, " : methode publique 3\n"
		print "    invocation de privee()\n"
		privee ()
	end

	def protegee ()
		print "Instance ", @numero, " : methode proteg‚e\n"
	end

	def privee ()
		print "Instance ", @numero, " : methode priv‚e\n"
	end
	
	public		:publique_1, :publique_2
	protected	:protegee
	private		:privee
end

instance_1 = Nouvelle . new (1)
instance_2 = Nouvelle . new (2)
instance_1 . publique_1 ()
print "\n"
instance_1 . publique_2 (instance_2)
print "\n"
instance_2 . publique_3 ()


